core/src/main/java/de/danoeh/antennapod/core/storage/DBWriter.java

```diff
@@ -185,6 +185,48 @@ private static boolean deleteFeedMediaSynchronous(
         });
     }
 
+    /**
+     * Remove the listed items and their FeedMedia entries.
+     */
+    @NonNull
+    public static Future<?> deleteFeedItems(@NonNull Context context, @NonNull List<FeedItem> items) {
+        return dbExec.submit(() -> {
+            DownloadRequester requester = DownloadRequester.getInstance();
+            List<FeedItem> queue = DBReader.getQueue();
+            List<FeedItem> removedFromQueue = new ArrayList<>();
+            for (FeedItem item : items) {
+                if (queue.remove(item)) {
+                    removedFromQueue.add(item);
+                }
+                if (item.getMedia() != null && item.getMedia().isDownloaded()) {
+                    deleteFeedMediaSynchronous(context, item.getMedia());
+                } else if (item.getMedia() != null && requester.isDownloadingFile(item.getMedia())) {
+                    requester.cancelDownload(context, item.getMedia());
+                }
+            }
+
+            PodDBAdapter adapter = PodDBAdapter.getInstance();
+            adapter.open();
+            if (!removedFromQueue.isEmpty()) {
+                adapter.setQueue(queue);
+            }
+            adapter.removeFeedItems(items);
+            adapter.close();
+
+            for (FeedItem item : removedFromQueue) {
+                EventBus.getDefault().post(QueueEvent.irreversibleRemoved(item));
+            }
+
+            // we assume we also removed download log entries for the feed or its media files.
```

Comment:
Are they actually deleted?

Reply:
In my tests it worked well. If I delete one of the MP3 files and press the feed update button, the corresponding item in the queue view disappears.

Reply:
Items of local feeds do not create download log entries. If this method is used to clean up normal feeds, we could be left with download log entries, I think. I know that it is currently not used for normal feeds, so this is not a critical problem. I was just wondering if this might cause orphan download log entries later when the method is maybe used for other items.

Reply:
How can I improve that? Do you have a hint?

Reply:
When deleting a feed, it also deletes the download log entries. Maybe you can borrow some code there. I am not completely sure if it is a good idea but when I look at that method (as a caller), I would expect it to do cleanup for normal feeds too (even if it is currently not used for normal feeds). So having the log entries deleted is more of a bonus feature. If you do not want to do it, you can keep it the way it is.

Reply:
I see your point. In the code it looks like this already happens, see the following call path:

1. The method `DBWriter.deleteFeedItems()` calls `adapter.removeFeedItems(items);` (line 213).
2. The mothod `PodDBAdapter.removeFeedItems()` calls `removeFeedItem()` (line 862).
3. The mothod `PodDBAdapter.removeFeedItem()` calls `removeFeedMedia()` (line 846).
4. The mothod `PodDBAdapter.removeFeedMedia()` deletes the download log entry from `TABLE_NAME_DOWNLOAD_LOG` (line 829).

Are there further entries in `TABLE_NAME_DOWNLOAD_LOG` that should be deleted when the method `DBWriter.deleteFeedItems()` is called?

To be honest, I'm not fully aware of all the data structures in AntennaPod but I'll try my best. :-)

Reply:
> The mothod PodDBAdapter.removeFeedMedia() deletes the download log entry from TABLE_NAME_DOWNLOAD_LOG

I did not know that, sorry. What about changing the comment's first sentence to something like "Deleting media also removes the download log entries"?

Reply:
Good idea. I just added that comment to make it more clear.

Reply:
Looks like the method introduces quite some code duplication of the method `deleteFeed`. Could you please make them share the common code (for example by extracting a common method `deleteFeedItemsSynchronous`)?

Reply:
I refactored the code as described. If `DBWriter.deleteFeedItemsSynchronous()` is called the feed items and the feed itself are deleted in separate database transactions now but I assume that's ok.